lib/rgb_matrix/engine.ex

```diff
@@ -188,15 +210,17 @@ defmodule RGBMatrix.Engine do
   @impl GenServer
   def handle_cast({:interact, led}, state) do
     {render_in, animation} = Animation.interact(state.animation, led)
-    state = schedule_next_render(state, render_in)
-    state = %State{state | animation: animation}
+
+    state =
+      %State{state | animation: animation}
+      |> schedule_next_render(render_in)
 
     {:noreply, %State{state | animation: animation}}
```

Comment:
oops, this is updating the state struct twice...

Reply:
Ohhh, that's just fine, nobody will notice! I wonder if the compiler optimized it away...